dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/ConsentAwareStorage.kt

```diff
@@ -57,6 +59,13 @@ internal class ConsentAwareStorage(
             TrackingConsent.NOT_GRANTED -> null
         }
 
+        val metric = internalLogger.startPerformanceMeasure(
```

Comment:
why create this outside the lock and not inside the synchronized block?

Reply:
> [!NOTE]
> The goal here is not to measure just the write operation, but also the delay between the moment we want to write, and the moment the data is actually written. Because write operations are queued in a single thread, we could create a bottleneck, and measure this delay here will help us find ways to improve it.